crankshaft-monitor/src/lib.rs

```diff
@@ -0,0 +1,33 @@
+//! Crate for monitoring crankshaft events.
+use std::net::SocketAddr;
+
+use proto::{Event, monitor_service_server::MonitorServiceServer};
+use server::CrankshaftMonitorServer;
+use tokio::sync::broadcast;
+
+pub mod proto;
+pub mod server;
+
+/// The main external Api to start the crankshaft monitor.
+pub async fn start_monitoring(
+    addr: SocketAddr,
+) -> Result<
+    (
+        broadcast::Sender<Event>,
+        tokio::task::JoinHandle<Result<(), tonic::transport::Error>>,
+    ),
+    Box<dyn std::error::Error>,
+> {
+    let (event_sender, event_receiver) = broadcast::channel(16);
+    let monitor_service = CrankshaftMonitorServer::new(event_receiver);
```

Comment:
I would try to differentiate the names `CrankshaftMonitorServer` and `MonitorServiceServer` a little more. Right now, it's confusing to me as to why we purportedly have a server that is then taken as an argument to another server.